clickhouse/restore-clickhouse.sh

```
@@ -0,0 +1,298 @@
+perform_restore() {
+    local volume_path="$1"
+    local backup_date="$2"
+    local backup_file="clickhouse-backup-${backup_date}.tar"
+    local temp_backup="/tmp/${backup_file}"
+
+    log "Starting restore from backup..."
+    log "Source: ${STORAGE_BOX_HOST}:${BACKUP_BASE_DIR}/${backup_file}"
+    log "Destination: $volume_path"
+
+    # Create a backup of current data (just in case)
+    local current_backup="${volume_path}.pre-restore-$(date '+%Y%m%d-%H%M%S')"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)